### PR TITLE
Addds possibility to remove sections from account edit page via layout.xml

### PR DESCRIPTION
**Pull request by sreichel**
Just add opposite method to `addLink()` method in 'Mage_Customer_Block_Account_Navigation' and `Mage_Sales_Block_Order_Info`. *(`addLink()` is only used in these classes)*

### CODE REVIEW

**Comment by sreichel** [comment]
Should be only one commit ...

File: app/code/core/Mage/Sales/Model/Order/Status.php [hunk 1]
```diff
@@ -27,8 +27,14 @@
...
 {
+    /**
+     * Model event prefix
+     *
+     * @var string
+     */
+    protected $_eventPrefix = 'sales_order_status';
 
-    protected function _construct()
+     protected function _construct()
```
Comment (colinmollenhour) [inline app/code/core/Mage/Sales/Model/Order/Status.php]: Oops, extra space..

File: app/code/core/Mage/Sales/Block/Order/Info.php [hunk 2]
```diff
@@ -78,6 +78,20 @@ public function addLink($name, $path, $label)
         return $this;
     }
 
+    /**
```
Comment (colinmollenhour) [inline app/code/core/Mage/Sales/Block/Order/Info.php]: This was already added in a separate PR.

File: app/code/core/Mage/Customer/Block/Account/Navigation.php [hunk 3]
```diff
@@ -50,6 +50,20 @@ public function addLink($name, $path, $label, $urlParams=array())
         return $this;
     }
 
+    /**
```
Comment (colinmollenhour) [inline app/code/core/Mage/Customer/Block/Account/Navigation.php]: Ditto.

File: app/code/core/Mage/Adminhtml/Block/Customer/Edit/Tab/Account.php [hunk 4]
```diff
@@ -33,6 +33,8 @@
  */
 class Mage_Adminhtml_Block_Customer_Edit_Tab_Account extends Mage_Adminhtml_Block_Widget_Form
 {
+    $_eventPrefix = 'adminhtml_customer_edit_tab_account';
```
Comment (colinmollenhour) [inline app/code/core/Mage/Adminhtml/Block/Customer/Edit/Tab/Account.php]: Missing "protected" keyword

File: app/code/core/Mage/Adminhtml/Block/Template.php [hunk 5]
```diff
@@ -34,6 +34,13 @@
 class Mage_Adminhtml_Block_Template extends Mage_Core_Block_Template
 {
     /**
+     * Block event prefix
+     *
+     * @var string
+     */
+    $_eventPrefix = 'adminhtml';
```
Comment (colinmollenhour) [inline app/code/core/Mage/Adminhtml/Block/Template.php]: Missing "protected" keyword.

File: app/code/core/Mage/Checkout/Model/Cart.php [hunk 6]
```diff
@@ -244,6 +244,8 @@ public function addProduct($productInfo, $requestInfo=null)
         $product = $this->_getProduct($productInfo);
         $request = $this->_getProductRequest($requestInfo);
 
+        Mage::dispatchEvent('checkout_cart_product_add_before', array('request' => $product', product' => $product));
```
Comment (colinmollenhour) [inline app/code/core/Mage/Checkout/Model/Cart.php]: Seems unrelated.

**Comment by sreichel** [comment]
Had already closed this. Thanks for missing "protected"!

**Comment by colinmollenhour** [comment]
Oops, didn't notice that..

Btw, you can force-push on your feature branches to remove unwanted commits and whatnot, such as after `git rebase -i` or `git commit --amend`. This way it is not necessary to open a separate PR.

File: app/code/core/Mage/Adminhtml/Block/Customer/Edit/Tab/Account.php (hunk 4 above)
Comment (LeeSaferite) [inline app/code/core/Mage/Adminhtml/Block/Customer/Edit/Tab/Account.php]: And, for consistency, the docblock should be added.

File: app/code/core/Mage/Adminhtml/Block/Template.php (hunk 5 above)
Comment (LeeSaferite) [inline app/code/core/Mage/Adminhtml/Block/Template.php]: Is this the correct event prefix for this class? It seems like `adminhtml_template` might be a better choice, n o?

File: app/code/core/Mage/Catalog/Model/Resource/Collection/Abstract.php [hunk 7]
```diff
@@ -36,6 +36,13 @@
 class Mage_Catalog_Model_Resource_Collection_Abstract extends Mage_Eav_Model_Entity_Collection_Abstract
 {
     /**
+     * Model event prefix
+     *
+     * @var string
+     */
+    protected $_eventPrefix = 'eav_catalog_collection_abstract';
```
Comment (LeeSaferite) [inline app/code/core/Mage/Catalog/Model/Resource/Collection/Abstract.php]: I disagree with this event name. There is no reason to have the `eav` part.

Comment (sreichel) [inline app/code/core/Mage/Catalog/Model/Resource/Collection/Abstract.php]: +1 will be changed :)

File: app/code/core/Mage/Adminhtml/Block/Template.php (hunk 5 above)
Comment (sreichel) [inline app/code/core/Mage/Adminhtml/Block/Template.php]: @LeeSaferite this event related edits shoulnt be part of this PR. So i closed it. But your right, should be changed.
